tests/common.py

```diff
@@ -153,6 +173,10 @@ def assertHoliday(self, *args):  # noqa: N802
         """Assert each date is a holiday."""
         self._assertHoliday("holidays", *args)
 
+    def assertIslamicNoEstimatedHoliday(self, *args):  # noqa: N802
```

Comment:
I'm planning to add support for the `..._no_estimated` parameter for all Eastern calendars of that type. There will be a lot more methods. ;)

Reply:
That's why I'm trying to convert this to syntactic sugar at the moment 👀

Reply:
Where is our #2352 solution... :)

Reply:
I'm trying to properly generate and init all interal "holidays_[insert]" i.e. "holidays_islamic_no_estimated_non_observed" combination first, then another one to handle its conversion to these variants:
```python
assertIslamicNoEstimatedNonObservedHoliday(self, *args)
assertIslamicNoEstimatedNonObservedHolidayDates(self, *args)
assertIslamicNoEstimatedNonObservedHolidayName(self, name, *args)
assertIslamicNoEstimatedNonObservedHolidays(self, *args)
assertIslamicNoEstimatedNoNonObservedHoliday(self, *args)
assertIslamicNoEstimatedNoNonObservedHolidayName(self, name, *args)
assertIslamicNoEstimatedNoNonObservedHolidays(self, *args)
```

Reply:
I think it's better to separate all this into dedicated PR, the changes are too large (and deserve their own line in CHANGES 🙂).

Reply:
Yeah, that could work - no need for all these huge refactors to hold back Nigerian holidays changes after all